packages/transit-vehicle-overlay/src/utils.js

```diff
@@ -0,0 +1,298 @@
+}
+
+/**
+ * will take an input object (e.g., probably a defaultProp representing a leaflet style),
+ * deep copy that object, and return back a new obj with the .color set
+ *
+ * @param color
+ * @return deep copied object with color set
+ */
+export function setColor(color, obj) {
+  const retVal = deep(obj);
+  retVal.color = color;
+  return retVal;
+}
+
+/** return map zoom value from leaflet */
+export function getZoom(leaflet, defZoom = 15) {
+  let retVal = defZoom;
+  try {
+    retVal = leaflet.map.getZoom();
+  } catch (e) {
+    console.log(e);
+  }
+  return retVal;
+}
+
+/** check 360 heading for a valid number */
+export function checkHeading(heading) {
+  if (heading === null || heading < 0 || heading >= 360) {
+    heading = 0;
+  }
+  return heading;
+}
+
+/**
+ * Checks if a parameter is actually a function.
+ * @param {*} fn The function to call.
+ * @returns fn if fn is a function, or a dummy function.
+ * TODO: copied from map ... should be in core-utils?
+ */
+export function callIfValid(fn) {
+  if (typeof fn === "function") return fn;
+  return () => {};
+}
+
+/**
+ * geojson uses [lon,lat] (e.g., [X, Y]) in representing coordinates
+ * this utility function reverses the point order to be [lat, lon] (or [Y, X])
+ *
+ * @return array of reversed points in the line geom
+ */
+export function reverseGeojsonPointsInGeom(geom) {
+  const revPoints = [];
+  for (let i = 0; i < geom.coordinates.length; i++) {
+    const c = geom.coordinates[i].reverse();
+    revPoints.push(c);
+  }
+  return revPoints;
+}
+
+/**
+ * OTP encodes polylines - this method will decode such geometries
+ * @param geom
+ * @return decoded polyline
+ */
+export function decodePolyline(geom) {
+  let retVal = geom;
+  if (geom && geom.points) {
+    retVal = polyline.decode(geom.points);
+  }
+  return retVal;
+}
+
+/**
+ * find a vehicle based on an id (either vehicle or trip id) within a list of vehicles
+ * @param vehicleList - list of vehciles to be searched
+ * @param queryId - either trip or vehcile id (string) to search list
+ * @param defVal - return value when not found or errors happen
+ */
+export function findVehicleById(vehicleList, queryId, defVal = null) {
+  let retVal = defVal;
+  try {
+    if (vehicleList && queryId) {
+      vehicleList.some(v => {
+        if (queryId === v.vehicleId || queryId === v.tripId) {
+          retVal = v;
+          return true;
+        }
+        return false;
+      });
+    }
+  } catch (e) {
+    console.log(e);
+  }
+  return retVal;
+}
+
+/**
+ * do 2 vehicle have the same tracking ids
+ * @param vehicleA
+ * @param vehicleB
+ * @returns boolean
+ */
+export function isTracked(vehicleA, vehicleB) {
+  let retVal = false;
+
+  try {
+    if (vehicleA && vehicleB) {
+      if (
+        vehicleA.vehicleId === vehicleB.vehicleId ||
+        vehicleA.tripId === vehicleB.tripId
+      ) {
+        retVal = true;
+      }
+    }
+  } catch (e) {
+    console.log(e);
+  }
+  return retVal;
```

Comment:
This could be rewritten into a one-liner:

```javascript
return !!vehicleA && !!vehicleB && (vehicleA.vehicleId === vehicleB.vehicleId || vehicleA.tripId === vehicleB.tripId);
```